src/components/KModal.js

```diff
@@ -68,7 +71,7 @@ export const KModal = ({ modalVisible = false, setModalVisible }) => {
           />
           <KSpacer height={30} />
           <KButton
-            text={isProcessing ? "Loading..." : "Recipe"}
+            text={isProcessing ? "Loading..." : buttonText}
             color={isProcessing ? Colors.gray : Colors.persian_red}
             onPress={() => {
```

Comment:
On `onPress` the modal is now calling the OpenAI API trying to find a recipe based on the text in the text input, this logic was implemented knowing the modal is going to be used just for the home screen. We should refactor this so the API is not called when we try to change the user name.